forc/src/cli/plugin.rs

```diff
@@ -63,6 +63,11 @@ fn is_executable(path: &Path) -> bool {
 
 /// Whether or not the given path points to a valid forc plugin.
 fn is_plugin(path: &Path) -> bool {
+    // This is to check that we do not match executables like `deps/forc-cd4662b67fa2a5c9`,
```

Comment:
+1 for the helpful comment

Reply:
Out of curiosity, where these `deps` coming from? I'm curious how they end up in your `PATH`?

I don't think it's feasible for us to filter out every possible unwanted directory of executables that a user might have careless placed in their `PATH`, however if this is particularly common for some reason I'm open to it.

I think the proper fix here is: Rather than `panic!`ing when failing to retrieve a description for a plugin, we should instead use an empty description, or perhaps some default string along the lines of "Unable to retrieve a description for this plugin."

Reply:
Thinking on this further, if we fail to retrieve a description, perhaps we can just skip the plugin altogether? We could instead update our definition of `forc` plugins to *require* a description in order to be enumerated under the `forc plugins` command?

Reply:
> Out of curiosity, where these `deps` coming from? I'm curious how they end up in your `PATH`?

They exist after a `cargo build` inside `target/debug/deps` folder. If `PATH` contains `sway/target/debug`, then it triggers this issue, since files are recursively enumerated when searching for plugins.

> I don't think it's feasible for us to filter out every possible unwanted directory of executables that a user might have carelessly/accidentally placed in their `PATH`, however if this is particularly common for some reason I'm open to it.

Agreed, but in this case it happens in a default build environment, so I thought it was worth dealing with it directly.

Might be worth noting these executables respond to `-h`:

```
~/dev/sway/target/debug$ deps/forc-cd4662b67fa2a5c9 -h
Usage: -h [OPTIONS] [FILTERS...]

Options:
        --include-ignored 
                        Run ignored and not ignored tests
        --ignored       Run only ignored tests
...
```

And that is why they are being considered as plugins. 

I think the most reliable fix here would be to make sure that forc plugins provide some kind of well-specified manifest (say with a `--manifest` flag which could return a JSON description). That would make sure there are no conflicts with other executables providing an `-h` flag.

For my first PR, however, I decided to take the easy route and just fix the panic, and skip them directly since I wasn't sure it was worth spending too much time on this.

If you think it's worth the time fixing this in a more solid way let me know.

Reply:
Thanks for the explanation @tritao!

> I think the most reliable fix here would be to make sure that forc plugins provide some kind of well-specified manifest (say with a --manifest flag which could return a JSON description).

That sounds like a great idea! But probably best for a future PR like you mention.

> They exist after a cargo build inside target/debug/deps folder. If PATH contains sway/target/debug, then it triggers this issue, since files are recursively enumerated when searching for plugins.

I'm still a little unsure of how `sway/target/debug` ends up in your `PATH` in this case? Do you normally manually add the target directory of projects you work on to your `PATH`?

I think the more conventional approach for having Rust bins available on `PATH` is to have `~/.cargo/bin` on the `PATH` - that way when `cargo install <crate>` is ran, the `<crate>` bin is available via `PATH`.

If you want the debug build to be available on your `PATH` while you're testing etc, another option is to use `cargo install --debug --path /path/to/crate`.

When I'm testing in Sway locally, I find it more useful to make a habit of running `forc` with the `cargo run` command to ensure I'm always using the local state, and to avoid the need to have two steps. E.g.

```sh
cargo run --bin forc -- build --path /path/to/forc/project
```

Reply:
I added the target dir to the `PATH` because I was just getting started and didn't know any better regarding the workflow.

Thanks for explaining your workflow with `cargo`, I can just use that going forward.